hawkmoth/parser.py

```diff
@@ -477,11 +495,15 @@ def _recursive_parse(domain, comments, errors, cursor, nest):
 
         return [ds]
 
-    elif cursor.kind in [CursorKind.STRUCT_DECL, CursorKind.UNION_DECL,
-                         CursorKind.ENUM_DECL]:
+    elif cursor.kind in [CursorKind.STRUCT_DECL,
+                         CursorKind.UNION_DECL,
+                         CursorKind.ENUM_DECL,
+                         CursorKind.CLASS_DECL,
+                         CursorKind.CLASS_TEMPLATE]:
 
         # Do not set the decl_name for anonymous symbols (empty spelling).
-        decl_name = name if cursor.spelling != '' else None
+        _, decl_name = _type_fixup(cursor)
```

Comment:
It scares me a bit that a lot of the paths now call `_type_fixup()` and I'm no longer quite sure why. It does a lot of things, and I'm completely uncertain if all of the things are needed in all of the paths.

Certainly the function pointer and array stuff is completely  unnecessary for this path.

I'm thinking smaller and more dedicated building blocks, so that changing `_type_fixup()` doesn't become insurmountable.

Reply:
Yup, this one is less defensible. I'm only adding a couple of lines, but the function is already pretty wild. We need some parts, but I should be able to split it in a decent way.